src/search/search_engines/eager_search.cc

```diff
@@ -20,6 +20,13 @@
 using namespace std;
 
 namespace eager_search {
+static shared_ptr<PruningMethod> get_pruning_method_from_options(const Options &opts) {
+    if (opts.contains("pruning")) {
+        return opts.get<shared_ptr<PruningMethod>>("pruning");
+    }
+    return nullptr;
+}
+
```

Comment:
A shorthand for this is `return opts.get<shared_ptr<PruningMethod>>("pruning", nullptr);`. Not sure if this needs a separate function `get_pruning_method_from_options` then.

Reply:
Ah, nice, I didn't know the get method could take a default value!